utils/python/CIME/SystemTests/erp.py

```diff
@@ -12,133 +12,50 @@
-            shutil.copy("env_build.xml", os.path.join("LockedFiles","env_build.xml"))
-
-            # update the case to use the new values
-            self._case.read_xml()
-
-            # Use the second executable that was created
-            exeroot = self._case.get_value("EXEROOT")
-            cime_model = CIME.utils.get_model()
-            exefile  = os.path.join(exeroot,"%s.exe"%(cime_model))
-            exefile2 = os.path.join(exeroot,"%s.ERP%d.exe"%(cime_model,run))
-            if (os.path.isfile(exefile)):
-                os.remove(exefile)
-            shutil.copy(exefile2, exefile)
-
-            case_setup(self._case, test_mode=True, reset=True)
-            stop_n      = self._case.get_value("STOP_N")
-            stop_option = self._case.get_value("STOP_OPTION")
-
-            if run == 1:
-                expect(stop_n > 2, "ERROR: stop_n value %d too short"%stop_n)
-                rest_n = stop_n/2 + 1
-                self._case.set_value("REST_N", rest_n)
-                self._case.set_value("REST_OPTION", stop_option)
-                self._case.set_value("HIST_N", stop_n)
-                self._case.set_value("HIST_OPTION", stop_option)
-                self._case.set_value("CONTINUE_RUN", False)
-                suffix = "base"
-            else:
-                rest_n = stop_n/2 + 1
-                stop_new = stop_n - rest_n
-                expect(stop_new > 0, "ERROR: stop_n value %d too short %d %d"%(stop_new,stop_n,rest_n))
-                self._case.set_value("STOP_N", stop_new)
-                self._case.set_value("CONTINUE_RUN", True)
-                self._case.set_value("REST_OPTION","never")
-                suffix = "rest"
-            self.run_indv(suffix=suffix)
-
-        self._component_compare_test("base", "rest")
+        SystemTestsCompareTwo.__init__(self, case, True, run_one_st_archive = True)
+
+    def _case_one_setup(self):
+        pass
+
+    def _case_two_setup(self):
+        """ Case two uses half the number of threads and tasks as the defaults and case one. """
+        for comp in self._case.get_values("COMP_CLASSES"):
+            if comp == "DRV":
+                comp = "CPL"
+            ntasks = self._case.get_value("NTASKS_%s"%comp)
+            if ntasks > 1:
+                rootpe = self._case.get_value("ROOTPE_%s"%comp)
+                self._case.set_value("NTASKS_%s"%comp, ntasks // 2)
+                self._case.set_value("ROOTPE_%s"%comp, rootpe // 2)
+            nthreads = self._case.get_value("NTHRDS_%s"%comp)
+            if nthreads > 1:
+                self._case.set_value("BUILD_THREADED", True)
+                self._case.set_value("NTHRDS_%s"%comp, nthreads // 2)
+
+        rest_n = self._case.get_value("STOP_N") // 2 + 1
+        self._case.set_value("REST_N", rest_n)
+        self._case.set_value("REST_OPTION", self._case.get_value("STOP_OPTION"))
+
+        case_setup(self._case, test_mode=True, reset=True)
+
+    def _pre_run_one_hook(self):
```

Comment:
My point here is that these aren't symmetrical: There are legitimate cases (like this one) where you need to do something in between run 1 and run 2. However, I think that anything you need to do before run1 can safely be done earlier in the process, in one of the existing hooks. Thus, adding that extra hook just adds an unnecessary decision for test developers.